corehq/apps/locations/bulk_management.py

```diff
@@ -696,6 +703,20 @@ def _validate_types_tree(self):
                 for code in e.affected_nodes
             ]
 
+        # Verify that deleted types are not referenced by other types via foreign key
+        for deleted_type in self.types_to_be_deleted:
+            for field_name in _get_location_type_foreign_key_fields_minus_parent():
```

Comment:
would you mind pulling this loop into a helper?  It's pretty complex and would be nice to see in isolation